From 90183b35889961774178cc332169c18591d0542e Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Mon, 10 Apr 2023 14:21:49 -0400
Subject: [PATCH] gdk/macos: fix builds on macOS before 10.13

Nothing profound here, just need to get some of the workarounds
into the right places.
---
 gdk/macos/gdkmacosclipboard-private.h  | 4 ----
 gdk/macos/gdkmacospasteboard-private.h | 4 ++++
 gdk/macos/gdkmacospasteboard.c         | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/gdk/macos/gdkmacosclipboard-private.h b/gdk/macos/gdkmacosclipboard-private.h
index 5679b13a36f..4866cf22e00 100644
--- a/gdk/macos/gdkmacosclipboard-private.h
+++ b/gdk/macos/gdkmacosclipboard-private.h
@@ -27,10 +27,6 @@
 
 G_BEGIN_DECLS
 
-#ifndef AVAILABLE_MAC_OS_X_VERSION_10_13_AND_LATER
-typedef NSString *NSPasteboardType;
-#endif
-
 #define GDK_TYPE_MACOS_CLIPBOARD (_gdk_macos_clipboard_get_type())
 
 G_DECLARE_FINAL_TYPE (GdkMacosClipboard, _gdk_macos_clipboard, GDK, MACOS_CLIPBOARD, GdkClipboard)
diff --git a/gdk/macos/gdkmacospasteboard-private.h b/gdk/macos/gdkmacospasteboard-private.h
index cac18a8fde5..cc7f6b8e475 100644
--- a/gdk/macos/gdkmacospasteboard-private.h
+++ b/gdk/macos/gdkmacospasteboard-private.h
@@ -26,6 +26,10 @@
 
 G_BEGIN_DECLS
 
+#ifndef AVAILABLE_MAC_OS_X_VERSION_10_13_AND_LATER
+typedef NSString *NSPasteboardType;
+#endif
+
 @interface GdkMacosPasteboardItemDataProvider : NSObject <NSPasteboardItemDataProvider>
 {
   GdkContentProvider *_contentProvider;
diff --git a/gdk/macos/gdkmacospasteboard.c b/gdk/macos/gdkmacospasteboard.c
index 0903c5352ac..b26248a9548 100644
--- a/gdk/macos/gdkmacospasteboard.c
+++ b/gdk/macos/gdkmacospasteboard.c
@@ -400,7 +400,7 @@ _gdk_macos_pasteboard_register_drag_types (NSWindow *window)
       gdk_content_formats_get_gtypes (formats, &n_gtypes);
 
       if (n_gtypes)
-        [ret addObject:NSPasteboardTypeURL];
+        [ret addObject:PTYPE(URL)];
 
       gdk_content_formats_unref (formats);
     }
-- 
GitLab

